onnxruntime/core/common/profiler.h

```diff
@@ -78,6 +78,17 @@ class Profiler {
                              const std::initializer_list<std::pair<std::string, std::string>>& event_args = {},
                              bool sync_gpu = false);
 
+  void EndTimeAndRecordEvent(EventCategory category,
```

Comment:
Do these new methods need to be public?

Reply:
As overriding alternatives, these method allows the consumer of profiler to log the duration of an op under different situation.